modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360AttachmentAdapterUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.sw360.utils;
+
+import org.eclipse.sw360.antenna.model.artifact.Artifact;
+import org.eclipse.sw360.antenna.model.artifact.ArtifactFactWithPayload;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactSourceFile;
+import org.eclipse.sw360.antenna.sw360.rest.resource.attachments.SW360AttachmentType;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SW360AttachmentAdapterUtils {
+
+    public static Map<Path, SW360AttachmentType> getAttachmentsFromArtifact(Artifact artifact) {
```

Comment:
This method will always return a Map with only one entry, correct?
You can use one liner
```suggestion
    public static Map<Path, SW360AttachmentType> getAttachmentsFromArtifact(Artifact artifact) {
        return getSourceFile(artifact)
                            .map(sourceFile -> Collections.singletonMap(sourceFile, SW360AttachmentType.SOURCE)
                            .orElse(Collections.EMPTY_MAP);
    }
```